hack/make-rules/vet.sh

```diff
@@ -47,7 +47,7 @@ done
 
 if [[ ${#targets[@]} -eq 0 ]]; then
   # Do not run on third_party directories or generated client code or build tools.
-  targets=$(go list -e ./... | egrep -v "/(build|third_party|vendor|staging|clientset_generated)/")
+  mapfile -t targets < <(go list -e ./... | grep -E -v "/(build|third_party|vendor|staging|clientset_generated)/")
```

Comment:
Ref: https://github.com/koalaman/shellcheck/wiki/SC2207

Reply:
We prefer not to use mapfile, as it's only bash 4+.

Reply:
It has been updated, please approve if it is correct.